packages/server/test/functional/status.test.ts

```diff
@@ -0,0 +1,220 @@
+  })
+
+  afterAll(async () => {
+    await app.destroy()
+  })
+
+  beforeEach(async () => {
+    app.caching.resetAll()
+  })
+
+  test('Get status of new conduit should be undefined', async () => {
+    const st = await status.get(state.conduit.id)
+    expect(st).toBeUndefined()
+  })
+
+  test('Create status', async () => {
+    const st = await status.create(state.conduit.id)
+
+    expect(validateUuid(st.conduitId)).toBeTruthy()
+    expect(st.numberOfErrors).toBe(0)
+    expect(st.initializedOn).toBeUndefined()
+    expect(st.lastError).toBeUndefined()
+
+    state.status = st
+  })
+
+  test('Get status', async () => {
+    const st = await status.get(state.conduit.id)
+
+    expect(st).toEqual(state.status)
+  })
+
+  test('List outdated should contain status', async () => {
+    const outdateds = await status.listOutdated(ms('1h'), 10, 10)
+
+    expect(outdateds).toEqual([state.status])
+  })
+
+  test('Create status again should fail', async () => {
+    await expect(status.create(state.conduit.id)).rejects.toThrow()
+  })
+
+  test('Create status for other conduit', async () => {
+    let st = await status.get(state.conduit2.id)
+    expect(st).toBeUndefined()
+
+    st = await status.create(state.conduit2.id)
+
+    expect(validateUuid(st.conduitId)).toBeTruthy()
+    expect(st.numberOfErrors).toBe(0)
+    expect(st.initializedOn).toBeUndefined()
+    expect(st.lastError).toBeUndefined()
+
+    state.status2 = st
+  })
+
+  test('List outdated should contain both status', async () => {
+    const outdateds = await status.listOutdated(ms('1h'), 10, 10)
+
+    expect(outdateds).toEqual([state.status, state.status2])
+  })
+
+  test('Add error', async () => {
+    await status.addError(state.conduit.id, new Error('err1'))
+
+    const st = await status.get(state.conduit.id)
+    expect(st!.numberOfErrors).toBe(1)
+    expect(st!.lastError?.startsWith('Error: err1')).toBeTruthy()
+
+    state.status = st
+  })
+
+  test('Add another error', async () => {
+    await status.addError(state.conduit.id, new Error('err2'))
+
+    const st = await status.get(state.conduit.id)
+    expect(st!.numberOfErrors).toBe(2)
+    expect(st!.lastError?.startsWith('Error: err2')).toBeTruthy()
+
+    state.status = st
+  })
+
+  test('Add multiple errors', async () => {
+    await status.addError(state.conduit.id, new Error('err3'))
+    await status.addError(state.conduit.id, new Error('err4'))
+    await status.addError(state.conduit.id, new Error('err5'))
+
+    const st = await status.get(state.conduit.id)
+    expect(st!.numberOfErrors).toBe(5)
+    expect(st!.lastError?.startsWith('Error: err5')).toBeTruthy()
+
+    state.status = st
+  })
+
+  test('List outdated should not contain errored conduit', async () => {
+    // conduit currently has 5 registered errors so it should no be listed
+    const maxAllowedErrors = 5
+    const outdateds = await status.listOutdated(ms('1h'), maxAllowedErrors, 10)
+
+    expect(outdateds).toEqual([state.status2])
```

Comment:
I think it would be best if you were to check that outdated does not contain errored conduits!

Reply:
This check does that and also checks that other stuff returned is correct so it's harder to have a false positive here